packages/wallet/src/services/AssetsTracker.ts

```diff
@@ -47,16 +50,23 @@ export const createAssetsTracker = (
         // Fetch asset metadata only for assets not already present in assetsMap
         map((assetIds) =>
           assetIds.map((assetId) => {
-            if (assetsMap.has(assetId)) {
-              return of(assetsMap.get(assetId));
+            const assetInfo = assetsMap.get(assetId);
+            if (assetInfo && isAssetInfoComplete(assetInfo)) {
+              return of(assetInfo);
             }
-            logger.debug(`Fetching metadata for asset ${assetId}`);
-            return assetService(assetId);
+            const stopPolling = new Subject<void>();
+            return intervalBackoff(retryBackoffConfig).pipe(
```

Comment:
[347bc46](https://github.com/input-output-hk/cardano-js-sdk/pull/464/commits/347bc467d864e1c109156d7f92a780eadaffc381)
- Added `pollUntil` option to coldObservable
- Updated `createAssetsService` to use this option
- Removed polling code from AssetsTracker
- Updated test to use mock `assetsProvider` instead of mock `assetsService`